pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
+func TestListUsersWildcards(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	tests := ListUsersTests{
+		{
+			name: "direct_relationship_wildcard",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:2", "viewer", "user:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "direct_relationship_wildcard_with_direct_relationships_also",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*,user]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "user:will"),
+				tuple.NewTupleKey("document:2", "viewer", "user:maria"),
+			},
+			expectedUsers: []string{"user:*", "user:will"},
+		},
+		{
+			name: "multiple_possible_wildcards_user_granularity",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user]
 
+			type document
+				relations
+					define viewer: [ group:*, user:*]`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "group:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "multiple_possible_wildcards_group_granularity",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "group",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user]
+
+			type document
+				relations
+					define viewer: [ group:*, user:*]`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "group:*"),
+			},
+			expectedUsers: []string{"group:*"},
+		},
+	}
+	tests.runListUsersTestCases(t)
+}
+
+func (testCases ListUsersTests) runListUsersTestCases(t *testing.T) {
+	storeID := ulid.Make().String()
+
+	for _, test := range testCases {
 		ds := memory.New()
 		defer ds.Close()
```

Comment:
It seems more recently we're trying to encourage more usage of `t.Cleanup` instead of `defer` in test functions. Can we switch this to `t.Cleanup(ds.Close)`?